pkg/antctl/raw/supportbundle/command_test.go

```diff
@@ -0,0 +1,344 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package supportbundle
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"testing"
+
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/rest"
+	cgtesting "k8s.io/client-go/testing"
+
+	"antrea.io/antrea/pkg/apis/crd/v1beta1"
+	antrea "antrea.io/antrea/pkg/client/clientset/versioned"
+	fakeclientset "antrea.io/antrea/pkg/client/clientset/versioned/fake"
+	"antrea.io/antrea/pkg/client/clientset/versioned/scheme"
+)
+
+var (
+	clientConfig = &rest.Config{
+		APIPath: "/supportbundle",
+		ContentConfig: rest.ContentConfig{
+			NegotiatedSerializer: scheme.Codecs,
+			GroupVersion:         &schema.GroupVersion{Group: "system.antrea.io", Version: "v1beta1"},
+		},
+	}
+	controllerInfo = v1beta1.AntreaControllerInfo{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "antrea-controller",
+		},
+		NodeRef: v1.ObjectReference{
+			Kind: "Node",
+			Name: "node-1",
+		},
+	}
+	node1 = v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "node-1",
+			ResourceVersion: "0",
+		},
+		Status: v1.NodeStatus{
+			Addresses: []v1.NodeAddress{
+				{
+					Type:    v1.NodeInternalIP,
+					Address: "172.16.0.0",
+				},
+			},
+		},
+	}
+	node2 = v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "node-1",
+		},
+	}
+	node3 = v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "node-3",
+			ResourceVersion: "0",
+		},
+		Status: v1.NodeStatus{
+			Addresses: []v1.NodeAddress{
+				{
+					Type:    v1.NodeInternalIP,
+					Address: "127.0.0.1",
+				},
+			},
+		},
+	}
+	agentInfo1 = &v1beta1.AntreaAgentInfo{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "antrea-agent-1",
+		},
+		APIPort: 0,
+		PodRef: v1.ObjectReference{
+			ResourceVersion: "0",
+		},
+		NodeRef: v1.ObjectReference{
+			Kind: "Node",
+			Name: "node-1",
+		},
+	}
+	agentInfo2 = &v1beta1.AntreaAgentInfo{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "antrea-agent-2",
+		},
+		APIPort: 0,
+		PodRef: v1.ObjectReference{
+			ResourceVersion: "0",
+		},
+		NodeRef: v1.ObjectReference{
+			Kind: "Node",
+			Name: "node-3",
+		},
+	}
+	nameList = []string{"node-1", "node-3"}
+)
+
+func TestCreateControllerClient(t *testing.T) {
+	cmd := &cobra.Command{}
```

Comment:
there is no need for a `cobra.Command` instance here. you should be using `ctx := context.Background()`.